homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,211 @@
+"""
+Climate on Zigbee Home Automation networks.
+For more details on this platform, please refer to the documentation
+at https://home-assistant.io/components/climate.zha/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components import zha
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate import DOMAIN
+from homeassistant.const import (
+    TEMP_CELSIUS, TEMP_FAHRENHEIT, ATTR_TEMPERATURE)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zha']
+ATTR_HEATING_POWER = 'heating_power'
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Zigbee Home Automation lights."""
+    discovery_info = zha.get_discovery_info(hass, discovery_info)
+    if discovery_info is None:
+        return
+
+    endpoint = discovery_info['endpoint']
+
+    try:
+        discovery_info['local_temp'] \
+            = yield from endpoint.thermostat['local_temp']
+    except (AttributeError, KeyError):
+        pass
+
+    # Clean data
+    for data in ('manufacturer', 'model'):
+        if isinstance(discovery_info.get(data), bytes):
+            discovery_info[data] = discovery_info.get(data)\
+                .split(b'\x00')[0].decode('utf-8')
+        if isinstance(discovery_info.get(data), str):
+            discovery_info[data] = discovery_info.get(data).split('\x00')[0]
+
+    if discovery_info['manufacturer'] == 'Stelpro' and \
+            discovery_info['model'] == 'ST218':
+        async_add_devices([ClimateST218(**discovery_info)],
+                          update_before_add=True)
+    else:
+        _LOGGER.warning("ZigBee Climate %s %s not supported yet",
+                        discovery_info['manufacturer'],
+                        discovery_info['model'])
+
+
+class ClimateST218(zha.Entity, ClimateDevice):
+    """Climage Stelpro ST218"""
+    _domain = DOMAIN
+
+    def __init__(self, **kwargs):
+        """Initialize the ZHA Climate."""
+        super().__init__(**kwargs)
+
+        self._modes = {0: 'off',
+                       4: 'confort',
+                       5: 'eco',
+                       }
+        self._reverse_modes = dict((v, k) for k, v in self._modes.items())
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_operation = None
+        self._heating_power = None
+        self._operation_list = list(self._modes.values())
+        self._unit = 'C'
```

Comment:
Why do you set this only to translate it later into TEMP_CELSIUS? Can't you directly set TEMP_CELSIUS here?